src/unity/lib/unity_sframe.cpp

```diff
@@ -2016,27 +2196,39 @@ void unity_sframe::explore(const std::string& path_to_client, const std::string&
       }
 
       // parse the message as json
-      flex_int start = -1,
-               end = -1;
+      flex_int start = -1, end = -1, index = -1;
+      std::string column_name;
+
+      enum MethodType {GetRows = 0, GetModal = 1};
+      MethodType response;
+
       auto sa = gl_sarray(std::vector<flexible_type>(1, input)).astype(flex_type_enum::DICT);
       flex_dict dict = sa[0].get<flex_dict>();
       for (const auto& pair : dict) {
         const auto& key = pair.first.get<flex_string>();
         const auto& value = pair.second;
         if (key == "method") {
-          DASSERT_EQ(value.get<flex_string>(), "get_rows");
+          if(value.get<flex_string>() == "get_rows"){
```

Comment:
Interesting that method name and parameters are parsed in two different places. It may simplify the logic for sharing parameters between different methods, but looks a bit confusing (as it implies all methods can take all parameters). For now it's fine (since we only have two methods) but we may want to revisit this later to associate specific parameters with specific methods.